pnpm-lock.yaml

```diff
@@ -12,7 +12,7 @@ importers:
     dependencies:
       '@codemirror/autocomplete':
         specifier: ^6.17.0
-        version: 6.17.0(@codemirror/language@6.9.2)(@codemirror/state@6.2.1)(@codemirror/view@6.29.1)(@lezer/common@1.2.1)
+        version: 6.17.0(@codemirror/language@6.10.2)(@codemirror/state@6.4.1)(@codemirror/view@6.29.1)(@lezer/common@1.2.1)
```

Comment:
@gzdunek I've thought that pnpm solves the problem with deduping deps, but only now I've realized that it solves only the problem of incorrectly hoisted deps. ;f  https://github.com/pnpm/pnpm/issues/7258

Reply:
I thought the same too :|
We can add `pnpm dedupe --check` to the the "Install JS dependencies" step in GHA, although a config option as in the linked issue would be much better :/

Reply:
For there to not be a problem with deduping deps, pnpm would need to force a single version of a package through the whole workspace, which I'm not sure if it's even possible. The whole point of a workspace is that you can have different packages with different deps.

If you read [the in-depth explanation behind deduping in Yarn](https://yarnpkg.com/cli/dedupe#in-depth-explanation), it makes sense that deduping isn't run on every install. IMHO it should be ran when the packages change though, as this is the point where you can add duplication.

There's a whole thread in dependabot-core about [running dedupe only on a specific package with Yarn](https://github.com/dependabot/dependabot-core/pull/9388#issuecomment-2078625150).

I'd say let's let it rock for a little bit, see if it causes any issues. We could add `pnpm dedupe --check` to CI, but it's going to add some busywork that might turn out to not be that important. Compared to Yarn, where the duped deps were incorrectly hoisted, I believe the dupes should not be as big of a problem here.

Reply:
The monthly minor dep update does seem like a very good scenario to run dedupe. I just wish the bots were able to do this for us.

Reply:
Thanks for the links, it makes sense.